libs/arrow/build.gradle

```diff
@@ -0,0 +1,189 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+dependencies {
+  implementation project(':libs:opensearch-common')
+  implementation project(':libs:opensearch-core')
+  api "org.apache.arrow:arrow-vector:${versions.arrow}"
+  api "org.apache.arrow:arrow-format:${versions.arrow}"
+  api "org.apache.arrow:arrow-memory-core:${versions.arrow}"
+  runtimeOnly "org.apache.arrow:arrow-memory-netty-buffer-patch:${versions.arrow}"
+  runtimeOnly "org.apache.arrow:arrow-memory-netty:${versions.arrow}"
+  runtimeOnly "io.netty:netty-buffer:${versions.netty}"
+  runtimeOnly "io.netty:netty-common:${versions.netty}"
+
+  runtimeOnly "com.google.flatbuffers:flatbuffers-java:${versions.flatbuffers}"
+  implementation "org.slf4j:slf4j-api:${versions.slf4j}"
+  implementation "com.fasterxml.jackson.core:jackson-core:${versions.jackson}"
```

Comment:
Where the Jackson dependencies are coming from? I would expect only `arrow` related ones to be there? Many others like SnakeYAML, and  Blockhound (which I believe has to be in test scope) are not related, needed cleanup.

Reply:
SnakeYAML and Blockhound were added due to missing classes violations. I have added these violations into ignore list instead for forbidden APIs. 

Jackson is needed by `arrow-vector` for some of the annotations its using in few classes. I get following error i remove it - 
```
warning: unknown enum constant Include.NON_NULL
  reason: class file for com.fasterxml.jackson.annotation.JsonInclude$Include not found
/Users/rishma/.gradle/caches/modules-2/files-2.1/org.apache.arrow/arrow-vector/17.0.0/16685545e4734382c1fcdaf12ac9b0a7d1fc06c0/arrow-vector-17.0.0.jar(/org/apache/arrow/vector/types/pojo/Field.class): warning: Cannot find annotation method 'value()' in type 'JsonProperty'
/Users/rishma/.gradle/caches/modules-2/files-2.1/org.apache.arrow/arrow-vector/17.0.0/16685545e4734382c1fcdaf12ac9b0a7d1fc06c0/arrow-vector-17.0.0.jar(/org/apache/arrow/vector/types/pojo/Field.class): warning: Cannot find annotation method 'value()' in type 'JsonInclude'
warning: unknown enum constant Include.NON_EMPTY
error: warnings found and -Werror specified
```

Reply:
That would only need `jackson-annotations`

Reply:
@reta that's right, i was able to remove all jackson deps except `jackson-annotations` and `jackson-databind`. `jackson-databind` is needed for arrow-vectors at runtime.